hazelcast/reactor.py

```diff
@@ -92,14 +92,26 @@ def _cleanup_timer(self, timer):
 class AsyncoreConnection(Connection, asyncore.dispatcher):
     sent_protocol_bytes = False
 
-    def __init__(self, map, address, connection_closed_callback, message_callback):
+    def __init__(self, map, address, connect_timeout, socket_options, connection_closed_callback, message_callback):
         asyncore.dispatcher.__init__(self, map=map)
         Connection.__init__(self, address, connection_closed_callback, message_callback)
 
         self._write_lock = threading.Lock()
         self._write_queue = deque()
         self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+        # self.socket.settimeout(connect_timeout)
```

Comment:
looks like connect_timeout is not applied?

would it be possible to write a test for the timeout? I think it might be doable with the suspend_member() options in the RC